networking.md

```diff
@@ -0,0 +1,174 @@
+```
+
+## Making HTTP requests
+
+Next, create an HTTP [Client][client] with the `Client` constructor:
+
+<!-- skip -->
+```dart
+import 'package:http/http.dart' as http;
+
+var httpClient = new http.Client();
+```
+
+The client supports common HTTP operations, such as:
+
+* **HTTP GET:** Use [`get`][get] for a general request, [`read`][read] for a
+ request returning a string, or [`readbytes`][readbytes] for a request
+ returning bytes.
+
+* **HTTP POST:** Use [`post`][post] for a general post.
+
+Code sample:
+
+<!-- skip -->
+```dart
+postData() async {
+  ...
+  var response = await httpClient.post(url, body: {"name": "doodle", "color": "blue"});
+  print("Response status: ${response.statusCode}");
+}
+```
+
+Note that the HTTP APIs use [Dart
+Futures](https://www.dartlang.org/tutorials/language/futures) in the return
+values. We recommend using the API calls with the `async`/`await` syntax as shown in
+the preceding code sample.
+
+## Decoding and encoding JSON
+
+Support for decoding and encoding JSON is provided by the [`dart:convert`](https://docs.flutter.io/flutter/dart-convert/dart-convert-library.html) library.
+
+To decode the JSON string and parse the response into a Map:
+
+<!-- skip -->
+```dart
+Map data = JSON.decode(response.body);
+// Assume the response body is something like: ["foo", { "bar": 499 }]
+int barValue = data[1]['bar']; // barValue is set to 499
+```
+
+To encode JSON, pass a simple value (string, boolean, or number literal), or a
+Map, List, or List of Maps containing simple values, to the `encode` method:
+
+<!-- skip -->
+```dart
+String encodedString = JSON.encode([1, 2, { "a": null }]);
+```
+
+## Example: decoding JSON from HTTPS GET
+
+The following example shows how to decode JSON from an HTTPS GET call in a Flutter app.
+
+It calls the [httpbin.com](http://httpbin.com) web service testing API, 
+which then responds with your local IP address. Note that secure
+networking (HTTPS) is used.
+
+1. Create a new flutter app with `flutter create`.
+
+1. Add the [http dependency](#using-http-package).
+
+1. Replace the contents of `lib/main.dart` with the following:
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+
+void main() {
+  runApp(new MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      home: new MyHomePage(),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  MyHomePage({Key key}) : super(key: key);
+
+  @override
+  _MyHomePageState createState() => new _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  String _ipAddress = 'Unknown';
+
+  _getIPAddress() async {
+    String url = "https://httpbin.org/ip";
+    var httpClient = new http.Client();
+    var response = await httpClient.read(url);
+    Map data = JSON.decode(response);
+    String ip = data["origin"];
+
+    setState(() {
```

Comment:
Strictly speaking, it's not needed in this example.  However, people are likely to copy and paste this example code into situations where the check will be needed.  It's better to include the check for the general case to improve their copy/paste success rate.